spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkAction.java

```diff
@@ -53,7 +55,7 @@
 
 import static org.apache.iceberg.MetadataTableType.ALL_MANIFESTS;
 
-public abstract class BaseSparkAction<ThisT, R> implements Action<ThisT, R> {
+abstract class BaseSparkAction<ThisT, R> implements Action<ThisT, R> {
```

Comment:
This is also probably a good idea, not sure it's required for this PR

Reply:
Reverted for now.